enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestionResult.tsx

```diff
@@ -91,16 +103,26 @@ export const InteractiveQuestionResult = ({
       bg="var(--mb-color-bg-question)"
     >
       <Stack h="100%">
-        <Flex direction="row" gap="md" px="md" align="center">
+        <ViewHeaderContainer data-testid="qb-header" isNavBarOpen={false}>
+          {isOpenedFromDashboard && (
+            <DashboardBackButton noLink onClick={onNavigateBack} />
+          )}
+
           {withTitle &&
             (customTitle || (
               <h2 className={cx(CS.h2, CS.textWrap)}>
-                {question.displayName()}
+                {isSaved ? (
+                  question.displayName()
+                ) : isNative ? (
+                  t`New question`
+                ) : (
+                  <QuestionDescription question={question} />
+                )}
```

Comment:
nit: if you want to avoid nested ternaries, you can also either move this into a `useMemo` or use `ts-pattern`.

in this case it's a single-variable condition, so it's already quite clear though.